frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPartition.jsx

```diff
@@ -333,8 +333,11 @@ class EmptyPartition extends React.Component {
 class Column extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { expanded: false };
+    this.state = { hover: false, expanded: false };
   }
+  mouseHover = hover => {
```

Comment:
Having to do this for styling reasons feels like a smell to me. I'd much prefer we do conditional styling for something like hover using a framework like `styled-components` instead of rolling it ourselves.

We could just create a wrapper div that encapsulates that and use it

```js
const DragWrapper = styled.div`
   padding: "12px 14px";
   box-shadow: 0 2px 3px ${lighten(colors["text-dark"], 1.5)};
   
   &:hover: {
       transition: all 300ms linear;
       box-shadow: 0 2px 5px ${lighten(colors["text-dark"], 1.3)};
   }
`
```

```js
/* I think there's a way to add className(s) to the styled component def above but I forgot the syntax so I'm just using them as normal props here 
*/ 
 return connectDropTarget(connectDragSource(<DragWrapper className="mb1 bordered rounded>...</DragWrapper>))
```

Reply:
Also FWIW, I think the `transition` property needs to be moved out of the `hover` pseudoclass for it to work.

Reply:
The problem with the div-wrapped version is I don't know enough CSS to fix the alignment of the grab handle. Help is welcomed, see PR #14538.